torch/fx/passes/infra/partitioner.py

```diff
@@ -0,0 +1,227 @@
+                    id = assignment[user_node]
+                    user_partitions[partitions_by_id[id]] = None
+                else:
+                    user_partitions[Partition(nodes=[user_node])] = None
+
+            # Filter out all the partitions that has dependency on other users
+            # TODO: find a better way to do this, rather than pair-wise comparision
+            user_partitions_list = list(user_partitions.keys())
+            for i in range(len(user_partitions_list)):
+                for j in range(i + 1, len(user_partitions_list)):
+                    pi = user_partitions_list[i]
+                    pj = user_partitions_list[j]
+                    dependency = self.__partition_depends_on(pi, pj)
+                    if dependency == 1 and pj in user_partitions:
+                        del user_partitions[pj]
+                    elif dependency == -1 and pi in user_partitions:
+                        del user_partitions[pi]
+
+            # We use the following rules for partition assignment:
+            # 1. If none of the candidates has been assigned to a partition, create a new partition
+            # 2. If there is one partition candidate, assign to the partition
+            # 3. If there are more than one partition candidates, assign current node to the first partition and
+            #    merge the other partitions with first partition, since user_partitions doesn't have depedency between
+            #    each other.
+
+            assigned_candidate_partition_ids = [partition.id for partition in user_partitions if partition.id is not None]
+
+            if len(assigned_candidate_partition_ids) == 0:
+                # create a new partition
+                assign(node, next(new_partition_id))
+            elif len(assigned_candidate_partition_ids) == 1:
+                id = assigned_candidate_partition_ids[0]
+                assign(node, id)
+            else:
+                # users are assigned to more than one partition, since user_partitions doesn't have
+                # dependency on each other, they can be fused into a single partition
+                id = assigned_candidate_partition_ids[0]
+                assign(node, id)
+
+                reassignment: Dict[Node, int] = {}
+                for other_id in assigned_candidate_partition_ids[1:]:
+                    for other_node in partitions_by_id[other_id].nodes:
+                        reassignment[other_node] = id
+                for other_node in reassignment:
+                    assign(other_node, id)
+
+        # post processing to re-assign "getitem" nodes into upstream partition
+        logger.debug("Reassigning getitem nodes to its producer node's partition...")
+        nodes_reassignment: Dict[Node, int] = {}
+        for node in self.graph_module.graph.nodes:
+            is_tuple_output = True
+            for user in node.users:
+                if user.op != "call_function" or \
```

Comment:
do we know if all the ops are converted to `call_function` ? When I was playing around with AOT autograd I saw that `x.sin()` will get registered as a `call_method` op -  do we need to support that as well?